pkg/deployments/tasks/20221113-euler-rebalanced-linear-pool/test/test.fork.ts

```diff
@@ -0,0 +1,320 @@
+  });
+
+  describe('generate lack of main token and rebalance', () => {
+    it('withdraw main tokens', async () => {
+      // We're going to withdraw enough man token to bring the Pool below its lower target, which will let us later
+      // rebalance.
+
+      const { cash } = await vault.getPoolTokenInfo(poolId, USDC);
+      const scaledCash = cash.mul(USDC_SCALING);
+      const { lowerTarget } = await pool.getTargets();
+
+      const exitAmount = scaledCash.sub(lowerTarget.div(3)).div(USDC_SCALING);
+
+      await vault.connect(holder).swap(
+        {
+          kind: SwapKind.GivenOut,
+          poolId,
+          assetIn: pool.address,
+          assetOut: USDC,
+          amount: exitAmount,
+          userData: '0x',
+        },
+        { sender: holder.address, recipient: holder.address, fromInternalBalance: false, toInternalBalance: false },
+        MAX_UINT256,
+        MAX_UINT256
+      );
+    });
+
+    itRebalancesThePool(LinearPoolState.MAIN_LACK);
+  });
+
+  describe('join below upper target and rebalance', () => {
+    it('deposit main tokens', async () => {
+      // We're going to join with few tokens, causing for the Pool to not reach its upper target.
+
+      const { lowerTarget, upperTarget } = await pool.getTargets();
+      const midpoint = lowerTarget.add(upperTarget).div(2);
+
+      const joinAmount = midpoint.div(100).div(USDC_SCALING);
+
+      await vault
+        .connect(holder)
+        .swap(
+          { kind: SwapKind.GivenIn, poolId, assetIn: USDC, assetOut: pool.address, amount: joinAmount, userData: '0x' },
+          { sender: holder.address, recipient: holder.address, fromInternalBalance: false, toInternalBalance: false },
+          0,
+          MAX_UINT256
+        );
+    });
+
+    itRebalancesThePool(LinearPoolState.BALANCED);
+  });
+
+  describe('exit above lower target and rebalance', () => {
+    it('withdraw main tokens', async () => {
+      // We're going to exit with few tokens, causing for the Pool to not reach its lower target.
+
+      const { lowerTarget, upperTarget } = await pool.getTargets();
+      const midpoint = lowerTarget.add(upperTarget).div(2);
+
+      const exitAmount = midpoint.div(100).div(USDC_SCALING);
+
+      await vault.connect(holder).swap(
+        {
+          kind: SwapKind.GivenOut,
+          poolId,
+          assetIn: pool.address,
+          assetOut: USDC,
+          amount: exitAmount,
+          userData: '0x',
+        },
+        { sender: holder.address, recipient: holder.address, fromInternalBalance: false, toInternalBalance: false },
+        MAX_UINT256,
+        MAX_UINT256
+      );
+    });
+
+    itRebalancesThePool(LinearPoolState.BALANCED);
+  });
+
+  describe('rebalance repeatedly', () => {
+    itRebalancesThePool(LinearPoolState.BALANCED);
+    itRebalancesThePool(LinearPoolState.BALANCED);
+  });
+
+  describe('rebalancer query protection', () => {
+    it('reverts with a malicious lending pool',async () => {
```

Comment:
Does Euler have the Lending Pool concept? I noticed that LENDING_POOL address and eUSDC address are the same

Reply:
I will need to rework this in the test to not cause confusion. Thanks for the hint.

Reply:
This appears in the other tests too and should be fixed everywhere.

Reply:
I updated the tests to include the eToken